app/alarm/ui/src/main/java/org/phoebus/applications/alarm/ui/tree/ItemConfigDialog.java

```diff
@@ -287,29 +313,43 @@ private void validateAndStore(final AlarmClient model, final AlarmTreeItem<?> it
 
         if (item instanceof AlarmClientLeaf) {
             final AlarmClientLeaf pv = new AlarmClientLeaf(null, item.getName());
+
+            boolean validEnableDate;
+            {
+                final LocalDateTime selected_enable_date = enabled_date_picker.getDateTimeValue();
+                final String relative_enable_date = relative_date.getValue();
+
+                if ((selected_enable_date != null)) {
+                    validEnableDate = pv.setEnabledDate(selected_enable_date);
+                } else if (relative_enable_date != null) {
+                    final TemporalAmount amount = TimeParser.parseTemporalAmount(relative_enable_date);
+                    final LocalDateTime update_date = LocalDateTime.now().plus(amount);
+                    validEnableDate = pv.setEnabledDate(update_date);
+                } else {
+                    pv.setEnabled(itemEnabled.get());
+                    validEnableDate = true;
+                }
+            }
+
+            if (!validEnableDate) {
+                Alert prompt = new Alert(Alert.AlertType.INFORMATION);
```

Comment:
Opening an alert while already in a dialog isn't ideal.
With more than one monitor, we might not know where the alert pops up.
With dialogs, we tend to use the dialog helper to position the dialog close to the button etc. which triggered it.
.. still, this is likely good enough and not worth a lot of extra code

Reply:
Frequently used pattern: decorate the widget holding malformed data, e.g. red border plus error message in the input field.
But that is more coding for sure...